### PR TITLE
Check member is defined before calling match

### LINKED ISSUE #614: Search is broken in some situations
The details are on https://github.com/helm/hub/issues/55

### PR DESCRIPTION
**Pull request by scottrigby**
Fixes #614

Alternative to #615 

For follow-up, this comment still applies https://github.com/helm/monocular/pull/615#issuecomment-472462431

Signed-off-by: Scott Rigby <scott@r6by.com>

### CODE REVIEW

**Comment by prydonius** [comment]
cc @Dean-Coakley

**Comment by prydonius** [comment]
@scottrigby do we know what chart repositories are missing some of the metadata that causes this to fail? Want to reproduce this with the right chart repository locally, to see if this fixes the issue.

**Comment by scottrigby** [comment]
@prydonius I don't know which could be causing the issue 😑 However, based on the error in https://github.com/helm/hub/issues/55#issuecomment-471666795 (which I also see), and looking at the code, this appears to be the only place where `match` is called on an object member (the others appear to be directly on objects) 🤷‍♂️ 

To reproduce, what about a chart that only lists `email` (not `name`) in the `maintainers` map? I'm not set up to test, so that's just a guess.

Also, could `chart.attributes.repo.name.match(re)` be broken because of these two hub PRs that don't include a `name` in `repos.yaml`?
- https://github.com/helm/hub/pull/52
- https://github.com/helm/hub/pull/56

**Comment by prydonius** [comment]
Thanks for the explanation @scottrigby, I agree that this should be a fairly safe change to make, so I'm happy to go ahead and merge this.

> Also, could chart.attributes.repo.name.match(re) be broken because of these two hub PRs that don't include a name in repos.yaml?

`repos.yaml` is purely metadata in the Hub repo, `repos-values.yaml` is what actually gets used in Monocular and this just links to the repo URL. The missing attributes are actually coming from the repo indexes which we don't easily have control over. https://github.com/helm/hub/pull/56 is still a worthy effort though to standardise that metadata.

File: frontend/src/app/shared/services/charts.service.ts [hunk 1]
```diff
@@ -72,21 +72,25 @@ export class ChartsService {
...
+  checkDefinedMatch(member, matcher) {
+    return member && member.match(matcher);
+  }
+
   arrayMatch(keywords: string[], re): boolean {
     if(!keywords) return false
 
     return keywords.some((keyword) => {
-      return !!keyword.match(re)
+      return this.checkDefinedMatch(!!keyword, re)
```
Comment (prydonius) [inline frontend/src/app/shared/services/charts.service.ts]: This was sending `true` to `checkDefinedMatch` and `true.match` is undefined. We actually want to send the string.

```suggestion
      return this.checkDefinedMatch(keyword, re)
```

Comment (scottrigby) [reply frontend/src/app/shared/services/charts.service.ts]: Ah good call… fixed!

**Comment by prydonius** [comment]
Thanks @scottrigby!!

**Pull request merged**